wiki/DEBUGGING.md

```diff
@@ -0,0 +1,128 @@
+# An Introduction to Debugging in Haskell
+
+## Introduction
+
+There is a general purpose debugging guide available [here](https://wiki.haskell.org/Debugging), but it
+is outdated and lacks information on how to navigate a modern debugger like gdb. This guide will look two main
+cases for debugging a program in Haskell. The first being printing in the code using
+[Debug.Trace](https://hackage.haskell.org/package/base-4.21.0.0/docs/Debug-Trace.html), a built-in package that great
+for working through smaller problems. The second will focus on tracing through the code using the
+[GHCi debugger](https://downloads.haskell.org/~ghc/8.10.3/docs/html/users_guide/ghci.html#the-ghci-debugger), which
+is very closely related in function to gdb, the debugger used in C in courses like CSC209 and CSC369. **Note:** If
+there are any issues or if any new commands, tools or features end up being useful, please update this guide
+accordingly!
+
+## Debug.Trace
+
+This library is a quick and effective way to print values of objects in your Haskell Program. Please refer to the
```

Comment:
This section should describe two functions: `trace` and `traceM`. Show the type signatures and explain where to use each one. You don't need to discuss other functions.